dml.go

```diff
@@ -39,8 +41,22 @@ func convertToSliceOfMaps(obj any) ([]map[string]any, error) {
 	return recordMap, nil
 }
 
-func doBatchedRequestsForCollection(auth authentication, method string, url string, batchSize int, recordMap []map[string]any) error {
-	var dmlErrors error
+func processSalesforceResponse(resp http.Response) ([]SalesforceResult, error) {
+	results := []SalesforceResult{}
+	responseData, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	jsonError := json.Unmarshal(responseData, &results)
+	if jsonError != nil {
+		return nil, jsonError
+	}
+
+	return results, nil
+}
+
+func doBatchedRequestsForCollection(auth authentication, method string, url string, batchSize int, recordMap []map[string]any) (*SalesforceResults, error) {
```

Comment:
Please consider passing a struct to control the behavior of this method

There are too many arguments